lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1769,7 +1769,15 @@ def __init__(
         self._relation_name = relation_name
         self._container = self._get_container(container_name)
         self._container_name = self._get_container_name(container_name)
-        self._log_files = log_files or []
+
+        if not log_files:
+            log_files = []
+        elif isinstance(log_files, str):
+            log_files = [log_files]
+        else:
+            log_files = list(log_files)
```

Comment:
I wonder if, since the presumptive fall-through case here is that it's _already_ a list, if we're gaining anything and we don't want to type check that it's actually a list (after checking whether it's a string), otherwise fail and log an error

Reply:
I'd probably disagree with "try to cast stuff and jam it into a list and assume that if no exceptions come up, it makes sense as unsanitized input". No reason not to just throw our hands up and return a meaningful message, lest some other random type make it through that without throwing an exception and we end up with a user who's very dependent on an accidental API which we could change.